src/ssl.c

```diff
@@ -29435,6 +29441,7 @@ int SetDhExternal(WOLFSSL_DH *dh)
         return WOLFSSL_FATAL_ERROR;
     }
 
+#if !defined(HAVE_FIPS) || (defined(HAVE_FIPS_VERSION) && (HAVE_FIPS_VERSION>2))
```

Comment:
Why is this gated by FIPSv2?  And why only this part of the function?  And why are we returning success to the user in FIPSv1 case??

Reply:
The `wolfSSL_PEM_read_bio_DHparams` function was missing from the nginx build using the client's FIPS configuration. When I included it in the FIPS build `SetDhInternal` was missing. When I included `SetDhInternal`, gcc reported that `pub` and `priv` are missing. These members of the `DhKey` struct are not available in the FIPS build but the rest of the struct is available so there is no reason why the entire function should be gated off in the FIPS build.

In short: `pub` and `priv` are not available in FIPS builds but the rest of the `DhKey` struct is available so only the part of function that deals with those members needs to be excluded.

Reply:
Other areas of the code appear to use  WOLFSSL_DH_EXTRA to gate those variables.  Shouldn't we use the same pattern here instead of FIPS defines?

Reply:
It does make more sense to use `WOLFSSL_DH_EXTRA`. I changed it to use that instead.